host/CMakeLists.txt

```diff
@@ -174,6 +174,8 @@ if (UNIX)
   target_compile_options(oehost PRIVATE
     -Wno-attributes -Wmissing-prototypes -fPIC ${PLATFORM_FLAGS})
   target_compile_definitions(oehost PRIVATE _GNU_SOURCE)
+  target_compile_options(oehost PUBLIC -fstack-protector-strong)
```

Comment:
Compiling the following code with -fstack-protector-strong -fPIE -O2 reveals that more work needs to be done:
void foo(void*);

void bar()
{
    int buffer[100];
    foo(buffer);
}
asm:
bar:
  subq $424, %rsp
  movq %fs:40, %rax
  movq %rax, 408(%rsp)
  xorl %eax, %eax
  movq %rsp, %rdi
  call foo@PLT
  movq 408(%rsp), %rax
  xorq %fs:40, %rax
  jne .L5
  addq $424, %rsp
  ret
.L5:
  call __stack_chk_fail@PLT